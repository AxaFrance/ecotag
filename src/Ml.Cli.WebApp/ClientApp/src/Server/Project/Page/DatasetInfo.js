import React from 'react';
import Table from '@axa-fr/react-toolkit-table';

export const DatasetInfo = ({ dataset }) => {
  return (
    <div>
      <h2>Dataset</h2>
      <Table>
        <Table.Header>
          <Table.Tr>
            <Table.Th>Name</Table.Th>
            <Table.Th>Type</Table.Th>
            <Table.Th>Classification</Table.Th>
            <Table.Th>Nb fichier</Table.Th>
            <Table.Th>Date de cr√©ation</Table.Th>
          </Table.Tr>
        </Table.Header>
        <Table.Body>
          <Table.Tr>
            <Table.Td>{dataset.name}</Table.Td>
            <Table.Td>{dataset.type}</Table.Td>
            <Table.Td>{dataset.classification}</Table.Td>
            <Table.Td>{dataset.numberFiles}</Table.Td>
            <Table.Td>{dataset.createDate}</Table.Td>
          </Table.Tr>
        </Table.Body>
      </Table>
    </div>
  );
};

export default DatasetInfo;
