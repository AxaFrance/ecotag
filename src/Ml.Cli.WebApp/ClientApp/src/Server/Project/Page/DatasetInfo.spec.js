import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render, waitForElement } from '@testing-library/react';
import {DatasetInfo} from './DatasetInfo';
import {BrowserRouter as Router} from "react-router-dom";

describe('DatasetInfo', () => {
    it('DatasetInfo render correctly', async () => {
        const dataset = {
            "id": "0001",
            "name": "Carte verte",
            "type": "Image",
            "classification": "Publique",
            "numberFiles": 300,
            "createDate": "30/10/2019"
        };
        const { asFragment, getByText } = render(<Router><DatasetInfo dataset={dataset} /></Router>);
        const messageEl = await waitForElement(() => getByText('Dataset'));
        expect(messageEl).toHaveTextContent(
            'Dataset'
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

