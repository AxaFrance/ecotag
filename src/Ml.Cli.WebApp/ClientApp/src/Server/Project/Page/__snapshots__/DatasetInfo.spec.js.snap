// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetInfo DatasetInfo render correctly 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Dataset
    </h2>
    <table
      class="af-table"
    >
      <thead
        class="af-table-thead"
      >
        <tr
          class="af-table__tr"
        >
          <th
            class="af-table__th"
          >
            Name
          </th>
          <th
            class="af-table__th"
          >
            Type
          </th>
          <th
            class="af-table__th"
          >
            Classification
          </th>
          <th
            class="af-table__th"
          >
            Nb fichier
          </th>
          <th
            class="af-table__th"
          >
            Date de cr√©ation
          </th>
        </tr>
      </thead>
      <tbody
        class="af-table__body"
      >
        <tr
          class="af-table__tr"
        >
          <td
            class="af-table__cell"
          >
            Carte verte
          </td>
          <td
            class="af-table__cell"
          >
            Image
          </td>
          <td
            class="af-table__cell"
          >
            Publique
          </td>
          <td
            class="af-table__cell"
          >
            300
          </td>
          <td
            class="af-table__cell"
          >
            30/10/2019
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
